### PR TITLE
View author profile from metasheet

### PR DESCRIPTION
**Pull request by bfollington**
Fixes https://github.com/subconsciousnetwork/subconscious/issues/634

- View author profile from memo viewer metasheet

Simulator Screenshot - iPhone 14 - 2023-09-27 at 14 25 00

### CODE REVIEW

File: xcode/Subconscious/Shared/Components/Detail/MemoViewerDetailMetaSheetView.swift [hunk 1]
```diff
@@ -14,7 +14,8 @@ struct MemoViewerDetailMetaSheetView: View {
     @Environment(\.dismiss) private var dismiss
     var state: MemoViewerDetailMetaSheetModel
     var send: (MemoViewerDetailMetaSheetAction) -> Void
-
+    var onViewAuthorProfile: () -> Void
```
Comment (gordonbrander) [inline xcode/Subconscious/Shared/Components/Detail/MemoViewerDetailMetaSheetView.swift:17]: I would like to funnel this through `send` so we don't have to call two things and pass two things, and everything goes through one path.

I think the right way to do this is:

- pass down a `ViewStore<MemoViewerDetailMetaSheetModel>`
- send up a new action `MemoViewerDetailMetaSheetAction.requestUserProfileDetail`
  - handle as a no-op in `MemoViewerDetailMetaSheetModel.update`
- Use cursor to tag into `MemoViewerDetailAction.requestUserProfileDetail` in 
- Use `onReceive` and `compactMap` to replay select actions from `Store<MemoViewerDetailMetaSheetModel>` on `notify`.
    - Map `MemoViewerDetailAction.requestUserProfileDetail` to an equivalent `MemoViewerDetailNotification`

This way only `MemoViewerDetailView` has to know about `notification`. The children are just consuming ordinary state or ViewStore.

WDYT?

File: xcode/Subconscious/Shared/Components/Detail/MemoViewerDetailMetaSheetView.swift [hunk 2]
```diff
@@ -45,6 +46,20 @@ struct MemoViewerDetailMetaSheetView: View {
                             item: state.shareableLink ?? ""
                         )
                         .disabled(state.shareableLink == nil)
+
```
Comment (gordonbrander) [inline xcode/Subconscious/Shared/Components/Detail/MemoViewerDetailMetaSheetView.swift:49]: There should be a `Divider()` here.

**Review by gordonbrander — CHANGES_REQUESTED** [review]
A couple changes requested.

File: xcode/Subconscious/Shared/Components/Detail/MemoViewerDetailMetaSheetView.swift (hunk 2 above)
Comment (bfollington) [reply xcode/Subconscious/Shared/Components/Detail/MemoViewerDetailMetaSheetView.swift:47]: Done

File: xcode/Subconscious/Shared/Components/Detail/MemoViewerDetailMetaSheetView.swift (hunk 1 above)
Comment (bfollington) [reply xcode/Subconscious/Shared/Components/Detail/MemoViewerDetailMetaSheetView.swift]: Yep, I like it. I believe I've implemented this as you explained above, let me know if I missed something.

**Review by gordonbrander — APPROVED** [review]
LGTM!

**Pull request merged**